2019-01-17  Jakub Jelinek  <jakub@redhat.com>

	* d-spec.cc (lang_specific_driver): Make -shared-libphobos
	the default rather than -static-libphobos.

--- gcc/d/d-spec.cc.jj	2019-01-01 12:37:49.502444257 +0100
+++ gcc/d/d-spec.cc	2019-01-17 17:09:45.364949246 +0100
@@ -408,9 +408,9 @@ lang_specific_driver (cl_decoded_option
   /* Add `-lgphobos' if we haven't already done so.  */
   if (phobos_library != PHOBOS_NOLINK)
     {
-      /* Default to static linking.  */
-      if (phobos_library != PHOBOS_DYNAMIC)
-	phobos_library = PHOBOS_STATIC;
+      /* Default to shared linking.  */
+      if (phobos_library != PHOBOS_STATIC)
+	phobos_library = PHOBOS_DYNAMIC;
 
 #ifdef HAVE_LD_STATIC_DYNAMIC
       if (phobos_library == PHOBOS_STATIC && !static_link)
